pkg/util/provider/driver/fake.go

```diff
@@ -25,6 +25,11 @@ import (
 	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/status"
 )
 
+// VMs ...
+type VMs map[string]string
+
+var fakeVMs = make(VMs)
```

Comment:
Can you please move the fakeVMs as a private field in the `FakeDriver` struct? This is possible in the OoT case this is possible because the driver is passed ot the controller instead of constructed by the controller.